File: app/src/main/java/com/github/overpass/gather/model/data/validator/UsernameValidator.java
package com.github.overpass.gather.model.data.validator;

import java.util.regex.Pattern;

public class UsernameValidator {

    public boolean isUsernameValid(String username) {
        return Pattern.compile("[A-Za-z0-9_]+")
                .matcher(username)
                .matches();
    }
}

File: app/src/main/java/com/github/overpass/gather/model/data/validator/UsernameValidator.kt
package com.github.overpass.gather.model.data.validator

import java.util.regex.Pattern

class UsernameValidator : Validator<String> {

    override fun isValid(username: String): Boolean {
        return Pattern.compile("[A-Za-z0-9_]+")
                .matcher(username)
                .matches()
    }
}
